BDPointAndroidXamarinDemo/MainActivity.cs

```diff
@@ -85,84 +49,135 @@ protected override void OnCreate(Bundle savedInstanceState)
+            geoButton.Click += (sender, e) =>
+            {
+                if (geoButton.Checked)
+                    StartGeoTrigger();
+                else
+                    StopGeoTrigger();
+
+            };
+
+            bgGeoButton = FindViewById<ToggleButton>(Resource.Id.bgGeoTrigger);
+            bgGeoButton.Click += (sender, e) =>
+            {
+                if (bgGeoButton.Checked)
+                    StartBgGeoTrigger();
+                else
+                    StopGeoTrigger();
+
+            };
+
+            tempoButton = FindViewById<ToggleButton>(Resource.Id.tempo);
+            tempoButton.Click += (sender, e) =>
+            {
+                if (tempoButton.Checked)
+                    StartTempo();
+                else
+                    StopTempo();
 
             };
         }
 
-        private void startAuthentication()
+        private void StartInit()
         {
-            if (!serviceManager.IsBlueDotPointServiceRunning)
+            projectId = editTextProjectId.Text;
+            if (!serviceManager.IsBluedotServiceInitialized)
             {
-				/* Start the Bluedot Point Service by providing with the credentials and a ServiceStatusListener, 
-				 * the app will be notified via the status listener if the Bluedot Point Service started successful.
-				 * 
-				 * Parameters
-				 * packageName  The package name of your app created in the Bluedot Point Access
-				 * apiKey       The API key generated for your app in the Bluedot Point Access
-				 * userName     The user name you used to login to the Bluedot Point Access
-				 * listener     A Service Status Listener
-                 */
-                serviceManager.SendAuthenticationRequest("<ENTER_API_KEY>", this);
-
-                updateLog("Authenticating..");
+                serviceManager.Initialize(projectId, this);
+                UpdateLog("Initializing..");
             }
             else
             {
-                updateLog("Already Authenticated");
+                UpdateLog("Already Initialized");
             }
 
         }
 
-        public void stopService()
+        private void StartGeoTrigger()
         {
-            if (serviceManager != null)
-            {
-                if (serviceManager.IsBlueDotPointServiceRunning)
-                {
-                    //Call the method stopPointService in ServiceManager to stop Bluedot PointService
-                    serviceManager.StopPointService();
-                    updateLog("Logged Out");
-                }
-                else
-                {
-                    updateLog("Already Logged out");
-                }
 
-            }
+            GeoTriggeringService.Builder()
+               .InvokeNotification(CreateNotification())
+               .Start(this, this);
+        }
+
+        private void StartBgGeoTrigger()
+        {
+            GeoTriggeringService.Builder()
+                .Start(this, this);
```

Comment:
We should not be using 'this' for the result receiver, it should be inline.

Reply:
I tried inline but was not able to get it instead I have created separate listeners, please check and suggest if any other improvement can be done